simplecpp.cpp

```diff
@@ -3128,17 +3128,25 @@ void simplecpp::preprocess(simplecpp::TokenList &output, const simplecpp::TokenL
                             const bool par = (tok && tok->op == '(');
                             if (par)
                                 tok = tok->next;
+                            const bool systemheader = (tok && tok->op == '<');
+                            if (systemheader)
+                                tok = tok->next;
                             if (tok) {
                                 const std::string &sourcefile = rawtok->location.file();
-                                const bool systemheader = (tok->str()[0] == '<');
-                                const std::string header(realFilename(tok->str().substr(1U, tok->str().size() - 2U)));
+                                const std::string header(realFilename(systemheader ? tok->str() : (tok->str().substr(1U, tok->str().size() - 2U))));
```

Comment:
unfortunate there are not proper unit tests for this. I get the feeling that with the old code there could be a token like this: `<foo.h>` with your code you seem to assume that the tokens will be `<` , `foo.h` and `>`..
it is a while since this was written so I am not sure.

Reply:
The ```/``` character in ```<foo/foo.h>``` also didn't work...

Additionally I added support for evaluation of
```#define FOO __has_include(<foo/foo.h>)```